### PR TITLE
Introduce Buffer reset for setting length to zero

### PR DESCRIPTION
**Pull request by charleso**
/cc @markhibberd This is needed for https://github.com/ambiata/ivory/pull/619

### CODE REVIEW

File: mundane-bytes/src/main/scala/com/ambiata/mundane/bytes/Buffer.scala [hunk 1]
```diff
@@ -67,6 +65,9 @@ object Buffer {
     wrapArray(b.bytes, b.offset + o, b.length - o)
   }
 
+  def reset(b: Buffer): Unit =
```
Comment (markhibberd) [inline mundane-bytes/src/main/scala/com/ambiata/mundane/bytes/Buffer.scala]: Minor thing, but for consistency should this return the `Buffer`, I don't actually care that much - and I kind of see why it is different, just thought I would ask to see if you had thought about it that much.

Comment (charleso) [inline mundane-bytes/src/main/scala/com/ambiata/mundane/bytes/Buffer.scala]: > , I don't actually care that much - and I kind of see why it is different, just thought I would ask to see if you had thought about it that much.

Yeah originally I had it return `Buffer`, but then it seemed like this type was more precise. Happy to change it if you prefer.

Comment (markhibberd) [inline mundane-bytes/src/main/scala/com/ambiata/mundane/bytes/Buffer.scala]: Really don't have a preference. Happy for it to be like this.

**Comment by markhibberd** [comment]
:vertical_traffic_light:

**Pull request merged**